cadc-access-control/src/main/java/ca/nrc/cadc/ac/client/UserClient.java

```diff
@@ -422,4 +431,136 @@ protected RegistryClient getRegistryClient()
         return new RegistryClient();
     }
 
+    /**
+     * Null safe method for a current Subject.
+     * @return  Subject instance.
+     */
+    private Subject getCurrentSubject()
+    {
+        final Subject subject = AuthenticationUtil.getCurrentSubject();
+
+        return (subject == null) ? new Subject() : subject;
+    }
+
+    /**
+     * Obtain the current AuthMethod instance from the current subject.
+     *
+     * The PASSWORD and TOKEN AuthMethods are NOT supported.
+     *
+     * @return      AuthMethod instance
+     * @throws AccessControlException       For unsupported AuthMethod.
+     */
+    private AuthMethod getAuthMethod()
+    {
+        final Subject subject = getCurrentSubject();
+        final Set<X509CertificateChain> x509CertificateChains =
+                subject.getPublicCredentials(X509CertificateChain.class);
+        final Set<SSOCookieCredential> cookieCredentials =
+                subject.getPublicCredentials(SSOCookieCredential.class);
+
+        // Prefer X509 Certificate.
```

Comment:
Sorry Dustin, another change:  could you please reverse this so that x509 credentials are preferred over cookie credentials?  That would mimic the way the other clients order their choices of auth method.